integration-tests/generate_rnode_data.py

```diff
@@ -0,0 +1,159 @@
+
+
+logging.getLogger('peers').setLevel(logging.INFO)
+logging.root.setLevel(logging.INFO)
+
+
+@contextlib.contextmanager
+def generate_cmd_opts() -> Generator[CommandLineOptions, None, None]:
+    yield CommandLineOptions(60 * 30, 60 * 30, 30, 60 * 30, None, None)
+
+
+def get_docker_folder(node: Node, target_dir: str, output_dir: str):
+    logging.info("Retrieve the data file from {} container".format(node.name))
+    if os.path.exists(output_dir):
+        shutil.rmtree(output_dir)
+    bits, _ = node.container.get_archive(target_dir)
+    with open(output_dir, 'wb') as f:
+        for chunk in bits:
+            f.write(chunk)
+
+def get_current_commit_hash():
+    p = subprocess.run(["git", "rev-parse", "HEAD"], capture_output=True, check=True)
+    with open("resources/rnode_data/version", 'wb') as f:
+        f.write(p.stdout)
+
+def generate_rnode_data() -> None:
+    """
+    This function would start a 3 nodes rchain network with one bootstrap node and two validators and start the genesis ceremony.
+    After the genesis ceremony is done, these node would try to propose to generate node data.
+    After the data generated above, it would copy the data to current directory which can be reused to start an existing network.
+    """
+    bootstrap_cli_options = {
+        '--deploy-timestamp':   '1',
+        '--required-sigs':      '2',
+        '--duration':           '5min',
+        '--interval':           '10sec',
+    }
+    peers_cli_flags = set(['--genesis-validator'])
+    peers_cli_options = {
+        '--deploy-timestamp':   '1',
+        '--required-sigs':      '2',
+    }
+    peers_keypairs = [
+        VALIDATOR_A_PRIVATE,
+        VALIDATOR_B_PRIVATE,
+    ]
+    random_seed = Random()
+
+    bonding_map = {
+        CEREMONY_MASTER_PRIVATE: 100,
+        VALIDATOR_A_PRIVATE: 110,
+        VALIDATOR_B_PRIVATE: 90
+    }
+
+    wallet_map = {
+        CEREMONY_MASTER_PRIVATE: 1000,
+        VALIDATOR_A_PRIVATE: 1000,
+        VALIDATOR_B_PRIVATE: 2000
+    }
+    logging.info("start the genesis ceremony")
+
+    with generate_cmd_opts() as command_line_options ,\
+            docker_client_context() as  docker_cli, \
+            testing_context(command_line_options, random_seed, docker_cli, bootstrap_key=CEREMONY_MASTER_PRIVATE, peers_keys=peers_keypairs, validator_bonds_dict=bonding_map) as context, \
+            temporary_wallets_file(context.random_generator, wallet_balance_from_private_key=wallet_map) as wallets, \
+            ready_bootstrap(context=context, cli_options=bootstrap_cli_options, wallets_file=wallets) as ceremony_master, \
+            started_peer(context=context, network=ceremony_master.network, bootstrap=ceremony_master, name='validator-a', private_key=VALIDATOR_A_PRIVATE, wallets_file=wallets, cli_flags=peers_cli_flags, cli_options=peers_cli_options, synchrony_constraint_threshold=0.33) as validator_a, \
```

Comment:
Generated DAG consists of 1 genesis, then 1 each node propose a single block, then 3 REV transfers on ceremony master, then again each node propose a single block. So should be 22 blocks total, if there won't be blocks proposed at the same height. Are you sure this won't be the case? Would it be more reliable to bump value to > 2/3?